crates/cli/src/file_utils.rs

```diff
@@ -308,16 +313,65 @@ pub fn filter_files_by_diff_aware_info(
         .collect();
 }
 
+/// Generate a fingerprint for a violation that will uniquely identify the violation. The fingerprint is calculated
+/// as is
+///  SHA2(<file-location-in-repository> - <characters-in-directory> - <content-of-code-line> - <number of characters in line>)
```

Comment:
Why do we not use the rule ID?

Reply:
Because the violation does not contain the rule id.

Reply:
no, it's passed to the function (see the function definition).